fetch-in-react-practice/client/components/app.jsx

```diff
@@ -76,29 +76,26 @@ export default class App extends React.Component {
      * TIP: Be sure to SERIALIZE the updates in the body with JSON.stringify()
      * And specify the "Content-Type" header as "application/json"
      */
-    const todosArray = this.state.todos;
-    for (var i = 0; i < todosArray.length; i++) {
-      if (todosArray[i].todoId === todoId) {
-        const newObject = todosArray[i];
-        newObject.isCompleted = !newObject.isCompleted;
-      }
-    }
-
-    /*  fetch('/api/todos/{todoId}', {
+    const todoIndex = this.state.todos.findIndex(item => item.todoId === todoId);
+    const todoObject = this.state.todos[todoIndex];
+    todoObject.isCompleted = !todoObject.isCompleted;
+    fetch(`/api/todos/${todoId}`, {
       method: 'PATCH',
       headers: {
         'Content-Type': 'application/json'
       },
-      body: JSON.stringify(todoId)
+      body: JSON.stringify(todoObject)
     })
       .then(response => response.json())
       .then(data => {
-        // eslint-disable-next-line no-console
-        console.log('data:', data);
+        this.state.todos.splice(todoIndex, 1, data);
+        this.setState({
```

Comment:
In general, we don't want to modify any objects or arrays in state, but instead create a copy and modify the copy.

You can copy an array with the `.slice()` method

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice